sound/soc/sof/ipc4-topology.c

```diff
@@ -921,8 +922,17 @@ static int sof_ipc4_widget_setup_comp_process(struct snd_sof_widget *swidget)
 		goto err;
 	}
 
+	/*
+	 * Below function should be called first before we use swidget->module_info,
+	 * because swidget->module_info is setup in this function.
+	 */
+	ret = sof_ipc4_widget_setup_msg(swidget, &process->msg);
+	if (ret)
+		goto free_available_fmt;
+
 	cfg_size = sizeof(struct sof_ipc4_base_module_cfg);
-	if (process->payload_with_output_fmt)
+	fw_module = swidget->module_info;
+	if (!fw_module->ext_library && process->payload_with_output_fmt)
```

Comment:
is the first check really necessary? Isnt it enough to know whether the output_fmt should be added or not?

Reply:
I think we still need to add test to make sure process->payload_with_output_fmt is only set for basefw modules for safety. we can add this check at the beginning. comments?

Reply:
@aiChaoSONG why do we need this safety check? Let's say the tplg writer sets this token for a module that's not in the base fw, what would happen? Module init would fail and the fw would return an error gracefully right? So, I don't think we really need this.

Reply:
What I am thinking is to detect this early in the kernel, and print message if both are set, Yeah, we could let the fw fails if both are set, this also helps to save a commit, let's do this then.